selftest/spinlocks/README.md

```diff
@@ -0,0 +1,12 @@
+# Spinlocks
+
+https://lwn.net/Articles/776909/
+
+In BTF annotated BPF maps, we can use a bpf_spin_lock to allow for atomic reads/updates of values in BPF maps.
+
+It is NOT supported in programs of type:
+
+- BPF_PROG_TYPE_KPROBE
+- BPF_PROG_TYPE_TRACEPOINT
+- BPF_PROG_TYPE_PERF_EVENT
```

Comment:
It's not mentioned in the attached article, do you know why it is not supported in certain types of programs? Is it because with these events, the atomicity is already guaranteed? I'd appreciate if you have a pointer to the related documentation.

Reply:
I'm not entirely sure why, but you can see [here](https://elixir.bootlin.com/linux/latest/source/tools/testing/selftests/bpf/verifier/helper_restricted.c#L126) where I got this from.

According to the [patch](https://patchwork.kernel.org/project/netdevbpf/patch/20211113142227.566439-3-me@ubique.spb.ru/) that added those verifier tests, these program types can result in some locking issues.

Reply:
Thanks for the pointers. This is very useful.